low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -314,11 +324,15 @@ def validate(self):
             raise SetupError(1, msg)
         logger.info("%s - Validation done" % self.name)
 
-        if self.args.coverage and 'self' in self.args.plan[0]:
-            raise SetupError(errno.EINVAL,
-                             "%s - Argument validation failure. %s",
-                             self.name,
-                             "Code coverage can not be enabled with self tests.")
+        # Service restart is required for coverage.
```

Comment:
Log the error in case of validation failure.
And shift the #L325 after complete validation success stage

Reply:
done.